catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/CatalogRolloverAction.java

```diff
@@ -97,51 +96,54 @@ public CatalogRolloverAction(FilenameGenerator filenameGenerator, String filenam
 
     @Override
     public String toString() {
-        return "CatalogRolloverAction{" +
-                "catalogFramework=" + catalogFramework +
-                ", filenameTemplate='" + filenameTemplate + '\'' +
-                ", filenameGenerator=" + filenameGenerator +
-                ", parentMetacardUpdater=" + parentMetacardUpdater +
-                '}';
+        return "CatalogRolloverAction{" + "catalogFramework=" + catalogFramework
+                + ", filenameTemplate='" + filenameTemplate + '\'' + ", filenameGenerator="
+                + filenameGenerator + ", parentMetacardUpdater=" + parentMetacardUpdater + '}';
     }
 
     @Override
     public MetacardImpl doAction(MetacardImpl metacard, File tempFile)
             throws RolloverActionException {
 
-        Subject subject = context.getUdpStreamProcessor()
-                .getSubject();
+        synchronized (context.isParentDirty()) {
```

Comment:
future enhancement: Since in the two places you use this you only use the returned atomic boolean to lock on (and are always in a synchronized block when checking the isParentDirty) You could probably just lock on the `context` object and use a regular boolean for where you check it since the atomic boolean isn't being taken advantage of.